pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -146,23 +179,66 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 	found := &kubevirtv1.VirtualMachine{}
 	err = r.client.Get(context.TODO(), types.NamespacedName{Name: vmSpec.Name, Namespace: vmSpec.Namespace}, found)
 	if err != nil && errors.IsNotFound(err) {
+
+		// Create kubevirt VM from oVirt VM:
 		reqLogger.Info("Creating a new VM", "VM.Namespace", vmSpec.Namespace, "VM.Name", vmSpec.Name)
+		err = r.client.Create(context.TODO(), vmSpec)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// Secret with username/password for the image import:
+		secretObj := &corev1.Secret{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Name: ovirtSecret, Namespace: vmSpec.Namespace}, secretObj)
+		if err != nil && errors.IsNotFound(err) {
+			dvSecret := createDVSecret(ovirtSecretDataMap, instance)
+			err = r.client.Create(context.TODO(), dvSecret)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+
+		// CM containing CA for the image import:
+		cm := &corev1.ConfigMap{}
```

Comment:
What do you think about extracting the whole block to a separate function (`ensureDVConfigmapExists`)?

Reply:
Yes, makes sense, thank you.